src/sentry/static/sentry/app/stores/repositoryStore.tsx

```diff
@@ -0,0 +1,88 @@
+import Reflux from 'reflux';
+
+import RepoActions from 'app/actions/repositoryActions';
+import {Repository} from 'app/types';
+
+type RepositoryStoreInterface = {
+  get(
+    orgSlug?: string
+  ): {
+    repos: Array<Repository> | undefined;
+    reposLoading: boolean | undefined;
+    reposError: Error | undefined;
+  };
+
+  orgSlug: string | undefined;
+  lastUpdated: number;
+  repos: Array<Repository> | undefined;
+  reposLoading: boolean | undefined;
+  reposError: Error | undefined;
+
+  loadRepos(orgSlug: string): void;
+  loadReposSuccess(data: Array<Repository>): void;
+  loadReposError(error: Error): void;
+};
+
+const RepositoryStoreConfig: Reflux.StoreDefinition & RepositoryStoreInterface = {
+  orgSlug: undefined,
+  lastUpdated: Date.now(), // withRepositories will refetch if data is older than 30s
+  repos: undefined,
+  reposLoading: undefined,
+  reposError: undefined,
+
+  listenables: RepoActions,
+
+  init() {
+    this.resetRepos();
+  },
+
+  resetRepos() {
+    this.orgSlug = undefined;
+    this.lastUpdated = Date.now();
+    this.repos = undefined;
+    this.reposLoading = undefined;
+    this.reposError = undefined;
+    this.trigger();
+  },
+
+  loadRepos(orgSlug: string) {
+    this.orgSlug = orgSlug;
+    this.lastUpdated = Date.now();
+    this.reposLoading = true;
+    this.trigger();
+  },
+
+  loadReposError(err: Error) {
+    this.lastUpdated = Date.now();
+    this.reposLoading = false;
+    this.reposError = err;
+    this.trigger();
+  },
+
+  loadReposSuccess(data: Array<Repository>) {
+    this.lastUpdated = Date.now();
+    this.reposLoading = false;
+    this.reposError = undefined;
+    this.repos = data;
+    this.trigger();
+  },
+
+  /**
+   * `organizationSlug` is optional. If present, method will run a check if data
+   * in the store originated from the same organization
+   */
+  get(orgSlug?: string) {
+    if (orgSlug && orgSlug !== this.orgSlug) {
```

Comment:
Should we be more explicit about the checking here, `orgSlug !== undefined && ...`?